Build failure on Raspberry Pi 4 (32bit)

<!--
        IMPORTANT: IF YOU DON'T FILL THIS TEMPLATE COMPLETELY IT WILL TAKE MORE TIME FOR US TO HELP YOU!
        SOME EXTERNAL ELECTRS GUIDES SUCH AS RASPIBOLT ARE OUTDATED AND DO NOT WORK SO SHOULD NOT BE FOLLOWED!
        Please try with OUR usage instructions first.

       If your build died with SIGKILL, try clearing up some RAM.
       If you have a low-memory device (such as RPi) try cross-compilation first.
-->

**Have you read the documentation?**
Yes. (Please, read usage.md first if you did not.)

**Did you double-check that you installed all dependencies?**
Yes. (Please, double check the dependencies if you didn't.)

**Which command failed?**
`cargo build`

**What was the error message?**

<details>
<summary>Error message</summary>

```
   Compiling bitcoincore-rpc v0.17.0
The following warnings were emitted during compilation:

warning: In file included from rocksdb/util/hash.h:26,
warning:                  from rocksdb/cache/sharded_cache.h:19,
warning:                  from rocksdb/cache/lru_cache.h:14,
warning:                  from rocksdb/cache/cache.cc:12:
warning: rocksdb/util/fastrange.h: In static member function ‘static Range rocksdb::detail::FastRangeGenericImpl<long long unsigned int, Range>::Fn(uint64_t, Range)’:
warning: rocksdb/util/fastrange.h:62:5: error: ‘__uint128_t’ was not declared in this scope
warning:      __uint128_t wide = __uint128_t{range} * hash;
warning:      ^~~~~~~~~~~
warning: rocksdb/util/fastrange.h:62:5: note: suggested alternative: ‘__uint32_t’
warning:      __uint128_t wide = __uint128_t{range} * hash;
warning:      ^~~~~~~~~~~
warning:      __uint32_t
warning: rocksdb/util/fastrange.h:62:45: error: invalid type argument of unary ‘*’ (have ‘uint64_t’ {aka ‘long long unsigned int’})
warning:      __uint128_t wide = __uint128_t{range} * hash;
warning:                                              ^~~~
warning: rocksdb/util/fastrange.h:63:31: error: ‘wide’ was not declared in this scope
warning:      return static_cast<Range>(wide >> 64);
warning:                                ^~~~
warning: rocksdb/util/fastrange.h:63:31: note: suggested alternative: ‘fwide’
warning:      return static_cast<Range>(wide >> 64);
warning:                                ^~~~
warning:                                fwide

error: failed to run custom build command for `electrs-librocksdb-sys v0.9.0-e2+7.8.3`

Caused by:
  process didn't exit successfully: `/home/admin/rust/electrs/target/release/build/electrs-librocksdb-sys-fc84180552873cd5/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=ROCKSDB_COMPILE
  cargo:rerun-if-env-changed=ROCKSDB_LIB_DIR
  cargo:rerun-if-env-changed=ROCKSDB_STATIC
  cargo:rerun-if-changed=rocksdb/
  TARGET = Some("armv7-unknown-linux-gnueabihf")
  OPT_LEVEL = Some("3")
  HOST = Some("armv7-unknown-linux-gnueabihf")
  cargo:rerun-if-env-changed=CXX_armv7-unknown-linux-gnueabihf
  CXX_armv7-unknown-linux-gnueabihf = None
  cargo:rerun-if-env-changed=CXX_armv7_unknown_linux_gnueabihf
  CXX_armv7_unknown_linux_gnueabihf = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = None
  cargo:rerun-if-env-changed=CXXFLAGS_armv7-unknown-linux-gnueabihf
  CXXFLAGS_armv7-unknown-linux-gnueabihf = None
  cargo:rerun-if-env-changed=CXXFLAGS_armv7_unknown_linux_gnueabihf
  CXXFLAGS_armv7_unknown_linux_gnueabihf = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = None
  cargo:rerun-if-env-changed=CXX_armv7-unknown-linux-gnueabihf
  CXX_armv7-unknown-linux-gnueabihf = None
  cargo:rerun-if-env-changed=CXX_armv7_unknown_linux_gnueabihf
  CXX_armv7_unknown_linux_gnueabihf = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = None
  cargo:rerun-if-env-changed=CXXFLAGS_armv7-unknown-linux-gnueabihf
  CXXFLAGS_armv7-unknown-linux-gnueabihf = None
  cargo:rerun-if-env-changed=CXXFLAGS_armv7_unknown_linux_gnueabihf
  CXXFLAGS_armv7_unknown_linux_gnueabihf = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = None
  running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-march=armv7-a" "-mfpu=vfpv3-d16" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "/home/admin/.cargo/registry/src/github.com-1285ae84e5963aae/zstd-sys-2.0.8+zstd.1.5.5/zstd/lib" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wshadow" "-Wno-unused-parameter" "-Wno-unused-variable" "-Woverloaded-virtual" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Wno-strict-aliasing" "-Wno-invalid-offsetof" "-std=c++17" "-DSNAPPY=1" "-DZSTD=1" "-DNDEBUG=1" "-DOS_LINUX" "-DROCKSDB_PLATFORM_POSIX" "-DROCKSDB_LIB_IO_POSIX" "-DROCKSDB_SUPPORT_THREAD_LOCAL" "-DHAVE_UINT128_EXTENSION=1" "-o" "/home/admin/rust/electrs/target/release/build/electrs-librocksdb-sys-f6c404ab82701ba9/out/rocksdb/cache/cache.o" "-c" "rocksdb/cache/cache.cc"
  cargo:warning=In file included from rocksdb/util/hash.h:26,
  cargo:warning=                 from rocksdb/cache/sharded_cache.h:19,
  cargo:warning=                 from rocksdb/cache/lru_cache.h:14,
  cargo:warning=                 from rocksdb/cache/cache.cc:12:
  cargo:warning=rocksdb/util/fastrange.h: In static member function ‘static Range rocksdb::detail::FastRangeGenericImpl<long long unsigned int, Range>::Fn(uint64_t, Range)’:
  cargo:warning=rocksdb/util/fastrange.h:62:5: error: ‘__uint128_t’ was not declared in this scope
  cargo:warning=     __uint128_t wide = __uint128_t{range} * hash;
  cargo:warning=     ^~~~~~~~~~~
  cargo:warning=rocksdb/util/fastrange.h:62:5: note: suggested alternative: ‘__uint32_t’
  cargo:warning=     __uint128_t wide = __uint128_t{range} * hash;
  cargo:warning=     ^~~~~~~~~~~
  cargo:warning=     __uint32_t
  cargo:warning=rocksdb/util/fastrange.h:62:45: error: invalid type argument of unary ‘*’ (have ‘uint64_t’ {aka ‘long long unsigned int’})
  cargo:warning=     __uint128_t wide = __uint128_t{range} * hash;
  cargo:warning=                                             ^~~~
  cargo:warning=rocksdb/util/fastrange.h:63:31: error: ‘wide’ was not declared in this scope
  cargo:warning=     return static_cast<Range>(wide >> 64);
  cargo:warning=                               ^~~~
  cargo:warning=rocksdb/util/fastrange.h:63:31: note: suggested alternative: ‘fwide’
  cargo:warning=     return static_cast<Range>(wide >> 64);
  cargo:warning=                               ^~~~
  cargo:warning=                               fwide
  exit status: 1
  running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-march=armv7-a" "-mfpu=vfpv3-d16" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "/home/admin/.cargo/registry/src/github.com-1285ae84e5963aae/zstd-sys-2.0.8+zstd.1.5.5/zstd/lib" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wshadow" "-Wno-unused-parameter" "-Wno-unused-variable" "-Woverloaded-virtual" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Wno-strict-aliasing" "-Wno-invalid-offsetof" "-std=c++17" "-DSNAPPY=1" "-DZSTD=1" "-DNDEBUG=1" "-DOS_LINUX" "-DROCKSDB_PLATFORM_POSIX" "-DROCKSDB_LIB_IO_POSIX" "-DROCKSDB_SUPPORT_THREAD_LOCAL" "-DHAVE_UINT128_EXTENSION=1" "-o" "/home/admin/rust/electrs/target/release/build/electrs-librocksdb-sys-f6c404ab82701ba9/out/rocksdb/cache/cache_entry_roles.o" "-c" "rocksdb/cache/cache_entry_roles.cc"
  exit status: 0

  --- stderr


  error occurred: Command "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-march=armv7-a" "-mfpu=vfpv3-d16" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "/home/admin/.cargo/registry/src/github.com-1285ae84e5963aae/zstd-sys-2.0.8+zstd.1.5.5/zstd/lib" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wshadow" "-Wno-unused-parameter" "-Wno-unused-variable" "-Woverloaded-virtual" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Wno-strict-aliasing" "-Wno-invalid-offsetof" "-std=c++17" "-DSNAPPY=1" "-DZSTD=1" "-DNDEBUG=1" "-DOS_LINUX" "-DROCKSDB_PLATFORM_POSIX" "-DROCKSDB_LIB_IO_POSIX" "-DROCKSDB_SUPPORT_THREAD_LOCAL" "-DHAVE_UINT128_EXTENSION=1" "-o" "/home/admin/rust/electrs/target/release/build/electrs-librocksdb-sys-f6c404ab82701ba9/out/rocksdb/cache/cache.o" "-c" "rocksdb/cache/cache.cc" with args "c++" did not execute successfully (status code exit status: 1).
```

</details>

**System**
Linux raspberrypi 5.10.103-v7l+ #1529 SMP Tue Mar 8 12:24:00 GMT 2022 armv7l GNU/Linux
rustc 1.73.0 (cc66ad468 2023-10-03)
cargo 1.73.0 (9c4383fb5 2023-08-26)

**Compilation**
Linking: static/dynamic
Cross compilation: no
Target architecture: armv71

**Additional context**
Any additional information that seems to be relevant.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,7 +40,7 @@ signal-hook = "0.3"
 tiny_http = { version = "0.12", optional = true }
 
 [dependencies.electrs-rocksdb]
-version = "0.19.0-e2"
+version = "0.19.0-e3"
 
 default-features = false
 # ZSTD is used for data compression

